### PR TITLE
UW-135: Adding all packages to environment.yaml.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Description**

Putting all packages in environment.yaml for full conda environment support.

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

I've tested the change locally on a Mac by installing an environment with:

```
conda env create -f environment.yaml 
conda activate uwtools
pip install .
pytest
```

I verified that the pip install step did not update or add any additional packages.

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: environment.yaml [hunk 1]
```diff
@@ -1,7 +1,14 @@
 name: uwtools
 channels:
+  - conda-forge
   - defaults
 dependencies:
+  - boto3=1.22*
+  - black
+  - f90nml=1.4*
```
[the remark this answers is not in the document]
Comment (christinaholtNOAA) [reply environment.yaml:8]: I don't think the environment.yml file allows for a "greater than" notation like that. This notation (documented here) gives us stability in "pinning" the major and minor release versions, but gives us the latest bug fixes at that version.

The stability helps to ensure that we aren't getting whiplash with new breaking changes that may creep in from our package dependencies until we are prepared to handle them.

[the remark this answers is not in the document]
Comment (WeirAE) [reply environment.yaml:8]: Fair, my thought was just to maintain consistency with setup.cfg. I think we should be able to use that based on [this documentation](https://docs.conda.io/projects/conda/en/4.6.0/_downloads/52a95608c49671267e40c689e0bc00ca/conda-cheatsheet.pdf) but limiting risks of surprise changes is at least as valuable.

[the remark this answers is not in the document]
Comment (WeirAE) [reply environment.yaml:8]: No further concerns, LGTM

**Review by WeirAE — DISMISSED** [review]
LGTM

**Review by WeirAE — APPROVED** [review]
So glad we can do this, LGTM!

**Pull request merged**